doc/decisions/1_in_discussion/copy_on_write.md

```diff
@@ -327,12 +324,101 @@ API notes:
 - `ksLookup()`:
   - if the `DEL` or `POP` flag is specified, do the COW-stuff as described multiple times now in the operations above
 
-#### Optimizations
+#### Reference Counting
+
+We need reference counting for the internal COW datastructures.
+For the beginning, we can just do it the same way reference counting currently works for `Key` and `KeySet`.
```

Comment:
Please avoid such statements: we don't want to do something "For the beginning" and later differently.

So we also need to decide reference counting first.